src/cracker.c

```diff
@@ -208,10 +205,10 @@ void crk_init(struct db_main *db, void (*fix_state)(void),
 	crk_last_key = crk_key_index = 0;
 	crk_last_salt = NULL;
 
-	if (fix_state)
-		(crk_fix_state = fix_state)();
-	else
-		crk_fix_state = crk_dummy_fix_state;
+	fix_state_present = fix_state || (options.flags & FLG_MASK_STACKED);
+
+	if ((crk_fix_state = fix_state))
+		crk_fix_state();
```

Comment:
I now reverted this and went back with the original `crk_dummy_fix_state()` instead of `fix_state_present`, because allowing `crk_fix_state` to be NULL would be error-prone in hybrid modes.